### PR TITLE
[test] Reinstate CoreGraphics↔︎NSValue bridging tests on iOS/watchOS/tvOS

### PR DESCRIPTION
**Pull request by lorentey**
This is a followup to #32502, reinstating a Foundation test.

### CODE REVIEW

**Comment by lorentey** [comment]
@swift-ci test

**Review by shahmishal — APPROVED** [review]
Thanks!

**Comment by shahmishal** [comment]
```
Failing Tests (1):
    Swift(windows-x86_64) :: SILGen/dso_handle.swift
```
cc: @compnerd

**Review by gribozavr — APPROVED** [review]

**Comment by lorentey** [comment]
Fascinating!  🤔

```
17:10:15 [ RUN      ] NSValueBridging.CGRect
17:10:15 stderr>>> 2020-06-23 17:10:15.320 NSValueBridging.swift.gyb.tmp.out[44542:1159733] +[NSValue valueWithCGRect:]: unrecognized selector sent to class 0x14bab18
17:10:15 stderr>>> *** [StdlibUnittest] Terminating due to uncaught exception NSInvalidArgumentException: +[NSValue valueWithCGRect:]: unrecognized selector sent to class 0x14bab18
17:10:15 stderr>>> libc++abi.dylib: terminating with uncaught exception of type NSException
17:10:15 stderr>>> CRASHED: SIGABRT
17:10:15 the test crashed unexpectedly
17:10:15 [     FAIL ] NSValueBridging.CGRect
```

**Comment by lorentey** [comment]
The iOS 10.3 simulator predates stable Swift ABI, so it doesn't have the UIKit overlay. Weak linking it fails, and since the executable isn't directly loading the UIKit framework, the method won't be available, either.

The 32-bit simulator tests are deploying the just-built stdlib & in-tree overlays to 10.3, which is not really useful at this point -- only the Swift 5 compatibility runtime will ever get deployed to iOS 10.3 in production, so that's what we should be testing.

**Comment by lorentey** [comment]
@swift-ci test

File: test/stdlib/NSValueBridging.swift.gyb [hunk 1]
```diff
@@ -18,11 +18,18 @@
 //
 // REQUIRES: objc_interop
 
+// The UIKit overlay isn't present on iOS 10.3.
+// UNSUPPORTED: CPU=i386
```
Comment (gribozavr) [inline test/stdlib/NSValueBridging.swift.gyb:22]: What about other tests that don't depend on UIKit? Could we put these tests into a separate file that does not run on the i386 iOS simulator, while the rest of the test does?

You could also use StdlibUnittest facilities to skip a test on the simulator. See these tests: https://github.com/apple/swift/blob/30ccb529485dac910a8f2a11504205fb5ea16873/validation-test/StdlibUnittest/Common.swift#L153

Comment (lorentey) [reply test/stdlib/NSValueBridging.swift.gyb:22]: That makes sense -- I'll submit a followup PR soon.

**Pull request merged**